docs/3.0/develop/transactions.mdx

```diff
@@ -119,3 +125,61 @@ def confirmation(transaction):
     print("committing a record now using the task's cache key!")
 ```
 </Tip>
+
+### Access data within transactions
+
+Key-value pairs can be set within a transaction and accessed elsewhere within the transaction, including within the `on_rollback` hook.
+
+The code below shows how to set a key-value pair within a transaction and access it within the `on_rollback` hook and the `quality_test` task: 
+
+```python
+import os
+from time import sleep
+
+from prefect import task, flow
+from prefect.transactions import get_transaction, transaction
+
+
+@task
+def write_file(contents: str):
+    "Writes to a file."
+    with open("side-effect.txt", "w") as f:
+        f.write(contents)
+
+
+@write_file.on_rollback
+def del_file(transaction):
+    "Deletes file."
+    os.unlink("side-effect.txt")
+    txn1 = get_transaction()      # get the transaction object
+    print(txn1.get("contents"))   # prints "hello world"
```

Comment:
with this example of a file being written / deleted, i think the use case for transaction key-value pairs is for the filename, which could be set from within the `write_file` task given that it's hardcoded there.